core/trino-main/src/main/java/io/trino/sql/analyzer/StatementAnalyzer.java

```diff
@@ -3671,6 +3702,11 @@ private Scope createScope(Optional<Scope> parentScope)
 
             return scopeBuilder;
         }
+
+        private OutputColumn createOutputColumn(Field field)
+        {
+            return new OutputColumn(new Column(field.getName().orElseThrow(), field.getType().toString()), analysis.getSourceColumns(field));
```

Comment:
Is it checked on some earlier stage of the analysis that all fields have names? If they don't, that should be a `SematicException` probably.

Reply:
We make sure that we create the ouputColumn after all the validation

Reply:
Indeed. So maybe just do an unchecked `get()`.

Reply:
From `Optional#get` java docs 
> The preferred alternative to this method is orElseThrow().